### DISCUSSION TITLE
Is there any attribute available to add on method for timing it?

### DISCUSSION THREAD

**Discussion by nicolaslattuada** [open-telemetry/opentelemetry-dotnet#4308] [Q&A] [answered]
Hello 
Is there something already available that allows to add metrics for execution time of a method via dotnet attribute?
What I am looking for could look like this:

```
class MyService {
    [Timed]
    public int MyLongMethod(int param)
    {
        // some action goes here
    }
}
``` 

And it would automatically record execution time into a metric like `execution.times.MyService.MyLongMethod`

Thanks

**Comment by cijothomas** [comment, marked as answer]
No, we do not have anything like that :(
 There was some discussions in the spec about a Timer instrument, but nothing exists today, so you'll have to manually report.

**Reply by nicolaslattuada** [reply]
ok Thanks !
